### PR TITLE
Project network layer

### LINKED ISSUE #192: Project NetworkLayer

### PR DESCRIPTION
**Pull request by arnekuchenbecker**

### CODE REVIEW

**Review by kittyahy — COMMENTED** [review]
When talking about an ID it's the offline Id, isn't  it?

**Review by kittyahy — CHANGES_REQUESTED** [review]
is there a possibility to test it? Or could you write a test?

File: app/src/main/java/com/scouts/kitchenplaner/networklayer/kitchenplaner/dtos/projects/ServerAllergenPeopleDTO.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+package com.scouts.kitchenplaner.networklayer.kitchenplaner.dtos.projects
+
+import java.util.Date
+
+/**
+ * DTO for communication with the server. Represents an allergic person.
+ *
+ * @param name The name of the person
+ * @param arrivalDate The date the person arrives
```
[the remark this answers is not in the document]
Comment (arnekuchenbecker) [reply app/src/main/java/com/scouts/kitchenplaner/networklayer/kitchenplaner/dtos/projects/ServerAllergenPeopleDTO.kt:25]: While that is true in principle,

1. From a kitchen perspective it really doesn't matter
2. Everywhere else we also wrote it this way, so for consistencies sake I would leave it like this

File: app/src/main/java/com/scouts/kitchenplaner/networklayer/ModelConversion.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+import com.scouts.kitchenplaner.networklayer.kitchenplaner.dtos.projects.ServerProjectDTO
+import com.scouts.kitchenplaner.networklayer.kitchenplaner.dtos.projects.ServerRecipeMappingDTO
+
+/**
+ * Creates a network layer DTO with the given onlineID from this project
+ *
+ * @param onlineID The onlineID of the project
+ * @return A network layer DTO representing this project
+ */
+fun Project.toNetworkLayerDTO(onlineID: Long): ServerProjectDTO {
```
[the remark this answers is not in the document]
Comment (arnekuchenbecker) [reply app/src/main/java/com/scouts/kitchenplaner/networklayer/ModelConversion.kt:35]: Fixed. I think I wrote this function before I was done with changing the ```Project``` class, which is also why there is no unit conversions and number changes being passed into this method. I created two issues to fix these (#204 and #205), as the unit conversions depend on that PR (#189) and the number changes require some thinking into how we represent those on both the server and the app as it is currently not elegantly possible to extract them from a project.

**Comment by arnekuchenbecker** [comment]
> When talking about an ID it's the offline Id, isn't it?

Depends on where you are, but usually in the app that is correct, yes

**Comment by arnekuchenbecker** [comment]
> is there a possibility to test it? Or could you write a test?

What exactly do you want to have tested? There is definitely the possibility to write tests for Retrofit API (I think I already did that for the Chefkoch stuff so it wouldn't be too difficult either), but I find it difficult to judge what functionality actually needs tests tbh. I think it would make sense to write at least an integration test once the model conversion for unit conversions (#204) and PersonNumberChanges (#205) are in, otherwise these test have to change once these issues are done.

**Review by kittyahy — APPROVED** [review]

**Pull request merged**